connector/src/main/java/com/google/cloud/hive/bigquery/connector/config/HiveBigQueryConfig.java

```diff
@@ -130,9 +133,21 @@
     // empty
   }
 
-  public static HiveBigQueryConfig from(Configuration conf, TableId tableId) {
+  public static HiveBigQueryConfig from(Configuration conf, java.util.Optional<TableId> tableId) {
     HiveBigQueryConfig config = new HiveBigQueryConfig();
-    config.tableId = tableId;
+    if (tableId.isPresent()) {
```

Comment:
```
config.tableId = tableId.orElse(TableId.of(
                  conf.get(HiveBigQueryConfig.PROJECT_KEY),
                  conf.get(HiveBigQueryConfig.DATASET_KEY),
                  conf.get(HiveBigQueryConfig.TABLE_KEY)));
```

Reply:
I've changed the code here slightly, so I'm not sure this format would apply any more. It will be replaced in a later PR anyways.